### ISSUE TITLE
[a11y-defect-2]: `va-table-inner` component has column headers scoped as "row" instead of "col"

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#95120]
### Point of contact

Trevor Pierce

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

The [VA table component](https://design.va.gov/components/table) header row tags are scoped as `th[scope="row"` instead of `th[scope="col"]`. This is preventing screen readers from announcing the column header as users traverse through tables in a non-linear fashion. I tested this with NVDA on Chrome and Firefox, and [confirmed the markup pattern](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/th) on MDN.

I have a working solution on my local machine that I will submit as a PR to the design system team after filing this issue.

### Reproduction steps

1. Navigate to the [VA table view](https://design.va.gov/components/table) on the design system site
2. Turn on the screen reader of your choice
3. Traverse through the table in a [non-linear fashion](https://webaim.org/articles/nvda/tables.htm). I prefer to drop down 2-3 rows, and then traverse across the row, one column at a time.
4. Verify you do not hear the column headers read aloud as you move across the row
...

### Proposed solution or next steps

I traced the issue to the `va-table-inner.tsx` file in the `makeRow()` function. I have an idea how to revise the logic cleanly so we can scope the table header columns correctly and maintain the correct markup for table rows. I'm proposing a refactor of lines 67-69:

```diff
! web-components/src/components/va-table/va-table-inner/va-table-inner.tsx

- return (i === 0 || row === 0)
-  ? <th scope="row">{slot}</th>
-  : <td>{slot}</td>

+ if (row === 0) {
+  return <th scope="col">{slot}</th>;
+ }

+ if (i === 0) {
+   return <th scope="row">{slot}</th>;
+ }

+ return <td>{slot}</td>;
```

### References, articles, or WCAG support

1. Understanding [SC 1.3.1](https://w3.org/TR/WCAG22#info-and-relationships): Info and Relationships (Level A)

### Type of issue

- [ ] Axe-core
- [x] Screenreader | assistive tech & device support
- [ ] Keyboard
- [ ] Zoom
- [ ] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [ ] Focus
- [ ] Headings
- [ ] Content style guide
- [ ] Link and button labels
- [x] Semantic HTML
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [x] Cognitive | user flows & navigating
- [ ] URL standards
- [ ] Something else
